### ISSUE TITLE
DatabasePropertyConfigResponse is a local type

### ISSUE THREAD

**Issue by aDogCalledSpot** [makenotion/notion-sdk-js#475]
**Describe the bug**
When fetching a database, there isn't a way to properly cast a property to the proper type it's supposed to be since these types aren't exported.

**To Reproduce**
Notion JS library version: current master

[Relevant line of code](https://github.com/makenotion/notion-sdk-js/blob/ba873383d5416405798c66d0b47fed3717c14f6a/src/api-endpoints.ts#L4808)

example:

```typescript
const response = await notion.databases.retrieve({ database_id });
const options = response.properties["My Select Property"].select; // Won't compile
```

**Expected behavior**
I would expect type safety when looking inspecting the properties in my database.

**Comment by aDogCalledSpot** [comment]
Same deal with `UpdateDatabaseBodyParameters`

**Comment by LoganTann** [comment]
Yep, it's very painful to work with these types unavailable, especially when using Typescript.

The same stands for `DatabaseObjectResponse`
